src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -477,5 +481,16 @@ static bool KeepAllItemsInTheList(CompletionTriggerKind initialTriggerKind, stri
                     initialTriggerKind == CompletionTriggerKind.Deletion;
             }
         }
+
+        public static async Task<SyntaxContext?> CreateSyntaxContextWithExistingSpeculativeModelAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            if (!document.SupportsSemanticModel)
```

Comment:
❔ Is this ever called on a document that doesn't support semantic models? It would be nice to be able to just throw if it doesn't, but we'd have to verify with TypeScript and F# before going with an approach like that.

Reply:
yeah, this seems odd to mbe.  i would throw here.   this should only be called from our own providers, which shoudl always succed onthis.